Sources/ArcGISToolkit/Components/OfflineMapAreasView/PreplannedMapModel.swift

```diff
@@ -16,22 +16,74 @@ import SwiftUI
 import ArcGIS
 
 /// An object that encapsulates state about a preplanned map.
-@MainActor
-class PreplannedMapModel: ObservableObject, Identifiable {
+public class PreplannedMapModel: ObservableObject, Identifiable {
     /// The preplanned map area.
     let preplannedMapArea: any PreplannedMapAreaProtocol
     
+    /// The view model for the map.
+    private var mapViewModel: OfflineMapAreasView.MapViewModel
+    
+    /// The task to use to take the area offline.
+    private let offlineMapTask: OfflineMapTask
+    
+    /// The download directory for the preplanned map areas.
+    private let preplannedDirectory: URL
+    
+    /// The ID of the preplanned map area.
+    private let preplannedMapAreaID: String
+    
+    /// The mobile map package for the preplanned map area.
+    private(set) var mobileMapPackage: MobileMapPackage?
+    
+    /// The currently running download job.
+    @Published private(set) var job: DownloadPreplannedOfflineMapJob?
+    
     /// The combined status of the preplanned map area.
     @Published private(set) var status: Status = .notLoaded
+
+    /// The result of the download job. When the result is `.success` the mobile map package is returned.
+    /// If the result is `.failure` then the error is returned. The result will be `nil` when the preplanned
+    /// map area is still packaging or loading.
+    @Published private(set) var result: Result<MobileMapPackage, Error>?
+    
+    /// A Boolean value indicating if download can be called.
+    var canDownload: Bool {
+        switch status {
+        case .notLoaded, .loading, .loadFailure, .packaging, .packageFailure,
+                .downloading, .downloaded:
+            false
+        case .packaged, .downloadFailure:
+            true
+        }
+    }
     
-    init(preplannedMapArea: PreplannedMapAreaProtocol) {
+    init?(
+        preplannedMapArea: PreplannedMapAreaProtocol,
+        mapViewModel: OfflineMapAreasView.MapViewModel,
+        mobileMapPackage: MobileMapPackage? = nil
+    ) {
+        offlineMapTask = mapViewModel.offlineMapTask
         self.preplannedMapArea = preplannedMapArea
+        self.mobileMapPackage = mobileMapPackage
+        preplannedDirectory = mapViewModel.preplannedDirectory
         
+        if let itemID = preplannedMapArea.id {
+            preplannedMapAreaID = itemID.rawValue
+        } else {
+            return nil
+        }
+        
+        self.mapViewModel = mapViewModel
```

Comment:
Be careful when holding reference to other objects. It seems each `MapViewModel` holds references to an array of `PreplannedMapModel`s, while each `PreplannedMapModel` holds a reference to the `MapViewModel`. That could lead to potential tight coupling issues.

Some of the shared properties don't need to come from the `MapViewModel`. For instance, the job manager is essentially a singleton, so it has a global `shared` reference; the directory URL can be passed as a value.

Reply:
Done. The reference to the map view model on each preplanned model has been removed.